resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -64,6 +64,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val MAX_SLAVE_FAILURES = 2
 
   private val maxCoresOption = conf.getOption("spark.cores.max").map(_.toInt)
+  private val maxMemOption = conf.getOption("spark.mem.max").map(Utils.memoryStringToMb)
```

Comment:
Can we defend against minimum values? For example default executor memory is 1.4MB. We could calculate the value returned by MesosSchedulerUtils.executorMemory. I don't think these values calculated in canLaunchTask ever change.

Reply:
@skonto 
For cpus, I think we can compare with minCoresPerExecutor
For mem, calling the MesosSchedulerUtils.executorMemory to get the minimum requirement.

Then at here, we parse the option, check the minimum and if it is too small, throw exception?

Reply:
Exception I think would ok, the idea if something is never going to work let the user know, especially for the novice user or the minimum would be a warning if we dont want an exception thrown.

Reply:
Should I add the check with this PR or a separate one?

Reply:
Do we need to have a check similar to https://github.com/apache/spark/blob/06df34d35ec088277445ef09cfb24bfe996f072e/resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala#L73 but for memory, so that we know we'll "land" on the maximum?